topology/src/main/java/io/camunda/zeebe/topology/serializer/ProtoBufSerializer.java

```diff
@@ -167,4 +177,57 @@ private Topology.State toSerializedState(
       case LEAVING -> Topology.State.LEAVING;
     };
   }
+
+  private Topology.ClusterChangePlan encodeChangePlan(final ClusterChangePlan changes) {
+    final var builder = Topology.ClusterChangePlan.newBuilder().setVersion(changes.version());
+    changes
+        .pendingOperations()
+        .forEach(operation -> builder.addOperation(encodeOperation(operation)));
+    return builder.build();
+  }
+
+  private Topology.TopologyChangeOperation encodeOperation(
+      final io.camunda.zeebe.topology.state.TopologyChangeOperation operation) {
+    final var builder =
+        Topology.TopologyChangeOperation.newBuilder().setMemberId(operation.memberId().id());
+    if (operation instanceof final PartitionJoinOperation joinOperation) {
+      builder
+          .setPartitionId(joinOperation.partitionId())
+          .setPriority(joinOperation.priority())
+          .setType(OperationType.PARTITION_JOIN_);
+    } else if (operation instanceof final PartitionLeaveOperation leaveOperation) {
+      builder.setPartitionId(leaveOperation.partitionId()).setType(OperationType.PARTITION_LEAVE);
+    }
```

Comment:
:wrench: let's add a default case that throws instead of silently not serializing operations.